providers/class-two-factor-backup-codes.php

```diff
@@ -305,7 +305,13 @@ public function authentication_page( $user ) {
 	 */
 	public function validate_authentication( $user ) {
 		$backup_code = isset( $_POST['two-factor-backup-code'] ) ? sanitize_text_field( wp_unslash( $_POST['two-factor-backup-code'] ) ) : false;
-		return $this->validate_code( $user, filter_var( $backup_code, FILTER_SANITIZE_STRING ) );
+		$success = $this->validate_code( $user, filter_var( $backup_code, FILTER_SANITIZE_STRING ) );
```

Comment:
I think we place logging logic in the core class where we validate the authentication provider: https://github.com/WordPress/two-factor/blob/c725c9b2affb3a0092427c09f544febb3f721ad9/class-two-factor-core.php#L1174

Are we getting benefits from adding logging logic here?

Reply:
We should not worry about the format of logging data or internal; the log function should dump logging data in an error log file.